staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/types.go

```diff
@@ -66,6 +66,32 @@ type ListOptions struct {
 	// it does not recognize and will return a 410 error if the token can no longer be used because
 	// it has expired.
 	Continue string
+
+	// SendInitialEvents, when set together with Watch option,
+	// begin the watch stream with synthetic init events to build the
+	// whole state of all resources followed by a synthetic "Bookmark"
+	// event containing a ResourceVersion after which the server
+	// continues streaming events.
+	//
+	// When SendInitialEvents option is set, we require ResourceVersionMatch
+	// option to also be set. The semantic of the watch request is as following:
+	// - ResourceVersionMatch = NotOlderThan
+	//   It starts with sending initial events for all objects (at some resource
+	//   version), potentially followed by an event stream until the state
+	//   becomes synced to a resource version as fresh as the one provided by
+	//   the ResourceVersion option. At this point, a synthetic bookmark event
+	//   is send and watch stream is continued to be send.
+	//   If RV is unset, this is interpreted as "consistent read" and the
+	//   bookmark event is send when the state is synced at least to the moment
+	//   when request started being processed.
+	// - ResourceVersionMatch = Exact
+	//   Unsupported error is returned.
+	// - ResourceVersionMatch unset (or set to any other value)
+	//   BadRequest error is returned.
```

Comment:
`ValidateListOptions` method returns `errors.NewInvalid` which sets `HTTP 422`. If that is okay with you I'll reword this comment.

Reply:
I'm fine with this, but I will leave it to API reviewers - @lavalamp I guess for this case.

Reply:
422 seems OK, especially if we use that elsewhere.